implementation/src/main/java/io/smallrye/mutiny/operators/uni/builders/UniJoinAll.java

```diff
@@ -95,6 +116,11 @@ private void forwardSignalWhenComplete() {
                 } else {
                     subscriber.onFailure(new CompositeException(failures));
                 }
+            } else if (concurrency != -1) {
+                int nextIndex = nextSubscriptionIndex.incrementAndGet();
```

Comment:
This should only happen if not canceled. After cancellation, we should not try to subscribe to a Uni.